### PR TITLE
(WIP) eq-367 Author create a survey

### PR DESCRIPTION
**Pull request by hamishtaplin**
### What is the context of this PR?

[Trello card](https://trello.com/c/i2ZKCSTR/367-8-author-create-a-survey)

Adds functionality for authors to add a survey and subsequently associate questionnaires with that survey.

### How to review 
Describe the steps required to test the changes.

- [ ] Do all commits have sensible commit messages?
- [ ] Is all new code unit tested?
- [ ] Are there integration tests if needed?
- [ ] Is there sufficient logging?
- [ ] Is there documentation or is the code self-describing?

### Who worked on the PR

Describe who worked on the changes, so that other people can review.

### CODE REVIEW

**Comment by hamishtaplin** [comment]
Currently, the `questionnaires` property of a survey is an array of ID—we need to add title, etc to be able to render the view.

File: dahl/assets/routes/Surveys/containers/SurveyListContainer/sagas/questionnaires.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import * as actions from './../actions'
+
+import request from 'utils/request'
+
+export function* addQuestionnaire(action) {
+  yield put(actions.addQuestionnaireRequest())
+  const response = yield call(request, '/api/v1/schema/', {
+    method: 'POST',
+    body: {
+      'mime_type': 'application/json/ons/eq',
```
Comment (hamishtaplin) [inline dahl/assets/routes/Surveys/containers/SurveyListContainer/sagas/questionnaires.js]: As discussed, the API should support adding new questionnaire without this stuff.

Comment (dhilton) [inline dahl/assets/routes/Surveys/containers/SurveyListContainer/sagas/questionnaires.js]: Which stuff specifically?

Comment (hamishtaplin) [inline dahl/assets/routes/Surveys/containers/SurveyListContainer/sagas/questionnaires.js]: @dhilton: the minimum data required to create a questionnaire and associate it with a survey should be just `title` and `survey_id`. The problem we have now is that, although sending this with the initial request works as far as creating a new Q is concerned, the backend doesn't support it elsewhere.

Basically, you can add a new Q and then navigate to it, save it and the API returns an error `{"detail":"Schema Failed Validation"}`.

**Comment by collisdigital** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;i2ZKCSTR&#x2F;367-8-author-create-a-survey"> (8) Author: Create a Survey</a>

**Pull request closed without merging**